pkg/maps/nat/nat.go

```diff
@@ -183,43 +183,47 @@ func (m *Map) Flush() int {
 	return int(doFlush6(m).deleted)
 }
 
-func deleteMapping4(m *Map, ctKey *tuple.TupleKey4Global) error {
-	key := *ctKey
+func deleteMapping4(m *Map, ctKey *ctmap.CtKey4Global) error {
+	key := NatKey4{
+		CtKey4Global: *ctKey,
+	}
 	// Workaround #5848.
 	addr := key.SourceAddr
 	key.SourceAddr = key.DestAddr
 	key.DestAddr = addr
 	valMap, err := m.Lookup(&key)
 	if err == nil {
-		val := *(*NatEntry4)(unsafe.Pointer(valMap.GetValuePtr()))
+		val := (*NatEntry4)(valMap.GetValuePtr())
 		rkey := key
 		rkey.SourceAddr = key.DestAddr
 		rkey.SourcePort = key.DestPort
 		rkey.DestAddr = val.Addr
 		rkey.DestPort = val.Port
-		rkey.Flags = tuple.TUPLE_F_IN
+		rkey.Flags = ctmap.TUPLE_F_IN
 
 		m.Delete(&key)
 		m.Delete(&rkey)
 	}
 	return nil
 }
 
-func deleteMapping6(m *Map, ctKey *tuple.TupleKey6Global) error {
-	key := *ctKey
+func deleteMapping6(m *Map, ctKey *ctmap.CtKey6Global) error {
+	key := NatKey6{
+		CtKey6Global: *ctKey,
+	}
 	// Workaround #5848.
 	addr := key.SourceAddr
 	key.SourceAddr = key.DestAddr
 	key.DestAddr = addr
 	valMap, err := m.Lookup(&key)
 	if err == nil {
-		val := *(*NatEntry6)(unsafe.Pointer(valMap.GetValuePtr()))
+		val := (*NatEntry6)(valMap.GetValuePtr())
```

Comment:
Is this no longer dereferencing the pointer because there is one less level of nesting in the structure?